File: Clases/Clase5/LSHADE/evaluate.cc
#include <math.h>

#define PI 3.1415926535897932384626433832795029

//  The same problema as before bur now with a constraint:
/**
double evaluate( double *vx, int n )
{
	double x = vx[0];
	double y = (x-2.0)*(x-5.0) + sin(1.5*PI*x);

	return( y );
}
**/

double evaluate( double *vx, int n )
{
	//double x = vx[0];
	double y = 0.0;
	for (int i = 0; i <= 9; i++) {
		y = y + (vx[i]-2.0)*(vx[i]-5.0) + sin(1.5*PI*vx[i]);
      }
	
	y = 0.1*y;

	return( y );
}


/**
double evaluate( double *vx, int n )
{
	double x = vx[0];
	double y = (x-2.0)*(x-5.0) + sin(1.5*PI*x);

	double g = y - 5.0 - 2.5*(x - 6.0);

	//  The constraint is introduced according the penalization method :

	double c = g;
	if ( g > 0 ) {  // The solution is feasible
 		c = 0.0;
	}
	return( y + 20.0*c*c );
}
**/

File: Clases/2020_Taller_Neo/Programas/Genetico/GA/evaluate.py
# coding: utf-8
#
# Here is defined the objective function
#
import math

# _n :  it is the number of variables 
# vpar : it is the vector with the values for each variable
#
def Evaluate( _n, vpar ) :
	x = vpar[0]
	v = (x-2)*(x-5) + math.sin( 1.5*math.pi*x )
	return v

# The same problema as before bur now with a constraint:
def Evaluate1( _n, vpar ) :

	x = vpar[0]
	y = (x-2.0)*(x-5.0) + math.sin(1.5*math.pi*x)

	g = y - 5.0 - 2.5*(x - 6.0)

	# The constraint is introduced according the penalization method :
	# 
	c = g
	if g > 0 :
		c = 0.0
	return( y + 20.0*c*c )



# La función esférica:
def Evaluate2( _n, vpar ) :
 	i=0
 	suma=0.0
 	while i<_n :
 		v = vpar[i]
 		suma += v*v
 		i += 1
 
 	return( suma )

# def Evalua( _n, vpar ):
# 	x = vpar[0]
# 
# 	r = (x-10.0)*(x-10.0) + 10.0 + 2.0 * math.cos( 2.0*x )
# 
# 	return r
